core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -109,7 +109,14 @@ private[spark] object ClosureCleaner extends Logging {
 
   private def createNullValue(cls: Class[_]): AnyRef = {
     if (cls.isPrimitive) {
-      new java.lang.Byte(0: Byte) // Should be convertible to any primitive type
+      cls match {
+        case java.lang.Boolean.TYPE => new java.lang.Boolean(false)
+        case java.lang.Character.TYPE => new java.lang.Character('\0')
+        case java.lang.Void.TYPE =>
+          // This should not happen because `Foo(void x) {}` cannot be compiled.
+          throw new IllegalStateException("This should not happen")
```

Comment:
I know this doesn't happen, but can we have a more explicit error message in case it does? e.g. "Unexpected void parameter in constructor"